File: go-sdk/podcastepisodes.go
package forem

import (
	"context"
	"fmt"
	"net/http"
	"openapi/pkg/models/operations"
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
	"strings"
)

type PodcastEpisodes struct {
	_defaultClient  HTTPClient
	_securityClient HTTPClient
	_serverURL      string
	_language       string
	_sdkVersion     string
	_genVersion     string
}

func NewPodcastEpisodes(defaultClient, securityClient HTTPClient, serverURL, language, sdkVersion, genVersion string) *PodcastEpisodes {
	return &PodcastEpisodes{
		_defaultClient:  defaultClient,
		_securityClient: securityClient,
		_serverURL:      serverURL,
		_language:       language,
		_sdkVersion:     sdkVersion,
		_genVersion:     genVersion,
	}
}

// GetPodcastEpisodes - Podcast Episodes
// This endpoint allows the client to retrieve a list of podcast episodes.
//
//	"Podcast episodes" are episodes belonging to podcasts.
//	It will only return active (reachable) podcast episodes that belong to published podcasts available on the platform, ordered by descending publication date.
//	It supports pagination, each page will contain 30 articles by default.
func (s *PodcastEpisodes) GetPodcastEpisodes(ctx context.Context, request operations.GetPodcastEpisodesRequest) (*operations.GetPodcastEpisodesResponse, error) {
	baseURL := s._serverURL
	url := strings.TrimSuffix(baseURL, "/") + "/api/podcast_episodes"

	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
	if err != nil {
		return nil, fmt.Errorf("error creating request: %w", err)
	}

	utils.PopulateQueryParams(ctx, req, request.QueryParams)

	client := s._securityClient

	httpRes, err := client.Do(req)
	if err != nil {
		return nil, fmt.Errorf("error sending request: %w", err)
	}
	defer httpRes.Body.Close()

	contentType := httpRes.Header.Get("Content-Type")

	res := &operations.GetPodcastEpisodesResponse{
		StatusCode:  int64(httpRes.StatusCode),
		ContentType: contentType,
	}
	switch {
	case httpRes.StatusCode == 200:
		switch {
		case utils.MatchContentType(contentType, `application/json`):
			var out []shared.PodcastEpisodeIndex
			if err := utils.UnmarshalJsonFromResponseBody(httpRes.Body, &out); err != nil {
				return nil, err
			}

			res.PodcastEpisodeIndices = out
		}
	case httpRes.StatusCode == 404:
	}

	return res, nil
}

File: java-sdk/lib/src/main/java/PodcastEpisodes.java
package ;

import .utils.HTTPClient;
import .utils.HTTPRequest;
import java.net.http.HttpResponse;
import com.fasterxml.jackson.databind.ObjectMapper;
import java.nio.charset.StandardCharsets;
import org.apache.http.NameValuePair;

public class PodcastEpisodes {
	private HTTPClient _defaultClient;
	private HTTPClient _securityClient;
	private String _serverUrl;
	private String _language;
	private String _sdkVersion;
	private String _genVersion;

	public PodcastEpisodes(HTTPClient defaultClient, HTTPClient securityClient, String serverUrl, String language, String sdkVersion, String genVersion) {
		this._defaultClient = defaultClient;
		this._securityClient = securityClient;
		this._serverUrl = serverUrl;
		this._language = language;
		this._sdkVersion = sdkVersion;
		this._genVersion = genVersion;
	}
	
	
    /**
     * getPodcastEpisodes - Podcast Episodes
     *
     * This endpoint allows the client to retrieve a list of podcast episodes.
     *         "Podcast episodes" are episodes belonging to podcasts.
     *         It will only return active (reachable) podcast episodes that belong to published podcasts available on the platform, ordered by descending publication date.
     *         It supports pagination, each page will contain 30 articles by default.
    **/
    public .models.operations.GetPodcastEpisodesResponse getPodcastEpisodes(.models.operations.GetPodcastEpisodesRequest request) throws Exception {
        String baseUrl = this._serverUrl;
        String url = .utils.Utils.generateURL(baseUrl, "/api/podcast_episodes");
        
        HTTPRequest req = new HTTPRequest();
        req.setMethod("GET");
        req.setURL(url);
        
        java.util.List<NameValuePair> queryParams = .utils.Utils.getQueryParams(request.queryParams);
        if (queryParams != null) {
            for (NameValuePair queryParam : queryParams) {
                req.addQueryParam(queryParam);
            }
        }
        
        HTTPClient client = this._securityClient;
        
        HttpResponse<byte[]> httpRes = client.send(req);

        String contentType = httpRes.headers().allValues("Content-Type").get(0);

        .models.operations.GetPodcastEpisodesResponse res = new .models.operations.GetPodcastEpisodesResponse() {{
            podcastEpisodeIndices = null;
        }};
        res.statusCode = Long.valueOf(httpRes.statusCode());
        res.contentType = contentType;
        
        if (httpRes.statusCode() == 200) {
            if (.utils.Utils.matchContentType(contentType, "application/json")) {
                ObjectMapper mapper = new ObjectMapper();
                mapper.findAndRegisterModules();
                .models.shared.PodcastEpisodeIndex[] out = mapper.readValue(new String(httpRes.body(), StandardCharsets.UTF_8), .models.shared.PodcastEpisodeIndex[].class);
                res.podcastEpisodeIndices = out;
            }
        }
        else if (httpRes.statusCode() == 404) {
        }

        return res;
    }
	
}